src/org/labkey/test/tests/AdminConsoleNavigationTest.java

```diff
@@ -71,7 +102,77 @@ public void testAdminNavTrails()
         }
 
         pagesMissingNavTrail.sort(Comparator.naturalOrder());
-        Assert.assertTrue("The following " + pagesMissingNavTrail.size() + " pages are missing Admin Console navtrails:\n" + String.join("\n", pagesMissingNavTrail), pagesMissingNavTrail.isEmpty());
+        assertTrue("The following " + pagesMissingNavTrail.size() + " pages are missing Admin Console navtrails:\n" + String.join("\n",
+                pagesMissingNavTrail), pagesMissingNavTrail.isEmpty());
+    }
+
+    @Test
+    public void testTroubleshooterLinkAccess()
+    {
+        ShowAdminPage adminConsole = goToAdminConsole();
+        impersonate(TROUBLESHOOTER);
+        Map<String, String> linkHrefs = new LinkedHashMap<>();
+        List<WebElement> troubleshooterLinks = adminConsole.getAllAdminConsoleLinks();
+        for (WebElement link : troubleshooterLinks)
+            linkHrefs.put(link.getText(), link.getAttribute("href"));
+
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (!link.getKey().equalsIgnoreCase("SignUp")           //Signup link shows up to the troubleshooter but throws 403 while accessing it.
+                    && !link.getKey().equalsIgnoreCase("Profiler")) //Profiler can be edited by the troubleshooter
+            {
```

Comment:
Creating a `Set.of()` ignoreLinks is going to be easier to maintain vs. multiple if conditions. Use the same pattern as the navtrail test above.

Reply:
Done